pkg/metrics/collector/kernel_metrics.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2018 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	tcpTimeoutDesc = prometheus.NewDesc(
+		"tcp_timeout_rehash",
+		"Tcp timeout rehash, collected from nstat",
+		nil, nil,
+	)
+	tcpDuplicateDesc = prometheus.NewDesc(
+		"tcp_duplicate_rehash",
+		"Tcp duplicate rehash, collected from nstat",
+		nil, nil,
+	)
+	segmentsReceivedDesc = prometheus.NewDesc(
+		"segments_received",
+		"Segments received on the node, collected from netstat -s",
+		nil, nil,
+	)
+	segmentsSentDesc = prometheus.NewDesc(
+		"segments_sent",
+		"Segments sent from node, collected from netstat -s",
+		nil, nil,
+	)
+	segmentsRetransmittedDesc = prometheus.NewDesc(
+		"segments_retransmitted",
+		"Segments retransmitted on the node, collected from netstat -s",
+		nil, nil,
+	)
+)
+
+type kernelStatCollector struct {
+}
+
+func init() {
+	registerCollector("kernel_metrics", NewKernelStatCollector)
+}
+
+func NewKernelStatCollector() (Collector, error) {
+	return &kernelStatCollector{}, nil
+}
+
+type kernelStats struct {
+	tcpTimeoutRehash      uint64
+	tcpDuplicateRehash    uint64
+	segmentsReceived      uint64
+	segmentsSent          uint64
+	segmentsRetransmitted uint64
+}
+
+func parseNetstatVals(stats *kernelStats, output string) error {
+	statcount := 0
+	for _, stat := range strings.Split(output, "\n") {
+		statList := strings.Fields(stat)
+		if len(statList) < 3 || statList[1] != "segments" {
+			continue
+		}
+		switch statList[2] {
+		case "received":
+			val, err := strconv.Atoi(statList[0])
+			if err != nil {
+				return fmt.Errorf("error casting to integer while getting segments received, err: %v", err)
+			}
+			statcount++
+			stats.segmentsReceived = uint64(val)
+		case "send":
```

Comment:
sent?

from net-tools v2.10-alpha, I see this TCP section which seems to have the right spellings..

Tcp:
    10124748 active connection openings
    6026748 passive connection openings
    11242 failed connection attempts
    7819340 connection resets received
    47 connections established
    502497809 segments received
    520382250 segments sent out
    15025 segments retransmitted
    7 bad segments received
    7847689 resets sent

Reply:
Strange I also have net-tools 2.10-alpha, but running it on alpine seems to give this output:

Tcp:
    0 active connections openings
    0 passive connection openings
    0 failed connection attempts
    0 connection resets received
    0 connections established
    0 segments received
    0 segments send out
    0 segments retransmited
    0 bad segments received.
    0 resets sent

Try running the netd container with net-tools installed and see what you get.